### PR TITLE
Remove redundant instructor queries in search #4229

### LINKED ISSUE #4229: Remove redundant instructor queries in search
`InstructorSearchPageAction`:
search_1
`*SearchQuery`:
search_2
`*SearchResultBundle`:
search_3
The same entity is queried three times.
The example is from `Comment*`, but all searches are guilty of this.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #4229

### CODE REVIEW

**Comment by JosephineKwa** [comment]
It looks good to merge :+1:

File: src/main/java/teammates/storage/api/StudentsDb.java [hunk 1]
```diff
@@ -46,14 +47,15 @@ public void putDocument(StudentAttributes student){
         putDocument(Const.SearchIndex.STUDENT, new StudentSearchDocument(student));
     }
     
-    public StudentSearchResultBundle search(String queryString, String googleId, String cursorString){
+    public StudentSearchResultBundle search(String queryString, String cursorString,
+                                            List<InstructorAttributes> instructorRoles) {
```
Comment (damithc) [inline src/main/java/teammates/storage/api/StudentsDb.java]: In some places, adding an extra parameter might make the method signature no longer self-explanatory. In those cases modify/add a header comment for the method.

**Comment by wkurniawan07** [comment]
Ready for review again.

**Comment by wkurniawan07** [comment]
@JosephineKwa can you review this by tmr afternoon? Thanks :smile:

**Comment by JosephineKwa** [comment]
Yup I'm reviewing it now! :)

**Pull request merged**